src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java

```diff
@@ -219,63 +205,33 @@ HtmlId forMember(TypeElement typeElement, VariableElement member) {
-     * @return the 1.4.x style anchor for the executable element
+     * @param executable the element to anchor to
      */
-    protected HtmlId forErasure(ExecutableElement executableElement) {
-        final StringBuilder buf = new StringBuilder(executableElement.getSimpleName().toString());
-        buf.append("(");
-        List<? extends VariableElement> parameters = executableElement.getParameters();
-        boolean foundTypeVariable = false;
-        for (int i = 0; i < parameters.size(); i++) {
-            if (i > 0) {
-                buf.append(",");
+    private String forErasure(TypeElement site, ExecutableElement executable) {
+        var parameterTypes = ((ExecutableType) utils.typeUtils
+                .erasure(utils.asInstantiatedMethodType(site, executable)))
+                .getParameterTypes();
+        var stv = new SimpleTypeVisitor9<String, Void>() {
+            @Override
+            public String visitArray(ArrayType t, Void p) {
+                return visit(t.getComponentType()) + utils.getDimension(t);
             }
-            TypeMirror t = parameters.get(i).asType();
-            SimpleTypeVisitor9<Boolean, Void> stv = new SimpleTypeVisitor9<>() {
-                boolean foundTypeVariable = false;
-
-                @Override
-                public Boolean visitArray(ArrayType t, Void p) {
-                    visit(t.getComponentType());
-                    buf.append(utils.getDimension(t));
-                    return foundTypeVariable;
-                }
-
-                @Override
-                public Boolean visitTypeVariable(TypeVariable t, Void p) {
-                    buf.append(utils.asTypeElement(t).getQualifiedName().toString());
-                    foundTypeVariable = true;
-                    return foundTypeVariable;
-                }
-
-                @Override
-                public Boolean visitDeclared(DeclaredType t, Void p) {
-                    buf.append(utils.getQualifiedTypeName(t));
-                    return foundTypeVariable;
-                }
-
-                @Override
-                protected Boolean defaultAction(TypeMirror e, Void p) {
-                    buf.append(e);
-                    return foundTypeVariable;
-                }
-            };
-
-            boolean isTypeVariable = stv.visit(t);
-            if (!foundTypeVariable) {
-                foundTypeVariable = isTypeVariable;
+
+            @Override
+            public String visitDeclared(DeclaredType t, Void p) {
+                return utils.getQualifiedTypeName(t);
             }
-        }
-        buf.append(")");
-        return foundTypeVariable ? HtmlId.of(buf.toString()) : null;
+
+            @Override
+            public String defaultAction(TypeMirror e, Void p) {
+                return String.valueOf(e);
+            }
+        };
+        return executable.getSimpleName() + parameterTypes.stream()
+                .map(stv::visit)
```

Comment:
Why do you need this (`stv::visit`) -- does the normal `.toString` for an array type not do the right thing?